crates/oci-distribution/src/lib.rs

```diff
@@ -145,6 +153,35 @@ impl Client {
             )),
         }
     }
+
+    /// Pull a single layer from an OCI registy.
+    ///
+    /// This pulls the layer for a particular image that is identified by
+    /// the given digest. The image reference is used to find the
+    /// repository and the registry, but it is not used to verify that
+    /// the digest is a layer inside of the image. (The manifest is
+    /// used for that.)
+    pub async fn pull_layer<T: std::io::Write>(
+        &self,
+        image: &Reference,
+        digest: &str,
+        mut out: T,
+    ) -> OciResult<T> {
+        let url = image.to_v2_blob_url(digest);
+        let mut stream = self
+            .client
+            .get(&url)
+            .headers(self.auth_headers())
+            .send()
+            .await?
+            .bytes_stream();
+
+        while let Some(bytes) = stream.next().await {
+            out.write_all(&bytes?.to_vec())?;
```

Comment:
Do you need the call to `to_vec`? I believe `bytes?.as_ref()` would work as well and avoids an allocation

Reply:
Fixed